starfish/core/spots/FindSpots/trackpy_local_max_peak_finder.py

```diff
@@ -0,0 +1,181 @@
+import warnings
+from functools import partial
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import xarray as xr
+from trackpy import locate
+
+from starfish.core.image.Filter.util import determine_axes_to_group_by
+from starfish.core.imagestack.imagestack import ImageStack
+from starfish.core.spots.FindSpots import spot_finding_utils
+from starfish.core.types import Axes, SpotAttributes, SpotFindingResults
+from ._base import FindSpotsAlgorithm
+
+
+class TrackpyLocalMaxPeakFinder(FindSpotsAlgorithm):
+    """
+    Find spots using a local max peak finding algorithm
+
+    This is a wrapper for :code:`trackpy.locate`, which implements a version of the
+    `Crocker-Grier <crocker_grier>`_ algorithm.
+
+    .. _crocker_grier: https://physics.nyu.edu/grierlab/methods3c/
+
+    Parameters
+    ----------
+
+    spot_diameter : odd integer or tuple of odd integers.
```

Comment:
1. params are not actually typed to match what is in the description.
2. the params' types in the documentation does not follow the standard style we use (which is whatever is the typing in the actual function declaration).
3. the styling for defaults does not match the styling for defaults in earlier PRs.
4. some defaults are missing.